Microsoft.DotNet.Interactive.CSharp/CSharpKernel.cs

```diff
@@ -12,23 +12,25 @@
 using Microsoft.CodeAnalysis.Completion;
 using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.CSharp.Scripting;
+using Microsoft.CodeAnalysis.QuickInfo;
 using Microsoft.CodeAnalysis.Recommendations;
 using Microsoft.CodeAnalysis.Scripting;
+using Microsoft.CodeAnalysis.Text;
 using Microsoft.DotNet.Interactive.Commands;
 using Microsoft.DotNet.Interactive.Events;
 using Microsoft.DotNet.Interactive.Extensions;
 using Microsoft.DotNet.Interactive.Formatting;
 using Microsoft.DotNet.Interactive.LanguageService;
 using Microsoft.DotNet.Interactive.Utility;
-using Newtonsoft.Json.Linq;
 using XPlot.Plotly;
 using Task = System.Threading.Tasks.Task;
 
 namespace Microsoft.DotNet.Interactive.CSharp
 {
     public class CSharpKernel :
         KernelBase,
-        IExtensibleKernel
+        IExtensibleKernel,
+        ISupportHover
```

Comment:
Another option here that might work is something like:

```csharp
public interface ILanguageServer<HoverText> { }
public interface ILanguageServer<Completion> { }
```

...etc.

The return types would also need to be represented somehow but it relates these different aspects of the overall set of LSP functionality. Maybe useful?

Reply:
`HoverText` would likely be specified like `class HoverText : ILanguageServiceParams<HoverParams, HoverResponse>`, but what about something like diagnostics, `Diagnostics : ILanguageServiceParams<void, DiagnosticsResult>`?  It feels better to specifically have `ISupportHover` that requires it's own very specific methods along with a separate (in the future) `ISupportDiagnostics` which will have a very different API due to the push nature, and wouldn't fit nicely into the `ILanguageServer<Diagnostics>` model.